addons/mail/static/src/models/chat_window/chat_window.js

```diff
@@ -178,6 +178,50 @@ function factory(dependencies) {
             return this.env._t("New message");
         }
 
+        /**
+         * @returns {boolean}
+         * @private
+         */
+        _computeIsVisible() {
+            if (!this.managerAllOrderedVisible) {
+                return false;
+            }
+            return this.managerAllOrderedVisible.includes(this);
+        }
+
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeThreadViewerShouldLoadMessages() {
+            if (this.threadViewerShouldLoadMessages) {
+                return this.threadViewerShouldLoadMessages;
+            }
+            if (!this.threadViewer) {
+                return false;
+            }
+            if (this.isVisible) {
+                this.threadViewer.update({ shouldLoadMessages: true });
+                return true;
+            }
+        }
+
+        /**
+         * @private
+         * @returns {integer|undefined}
+         */
+        _computeVisibleIndex() {
+            if (!this.manager) {
+                return undefined;
+            }
+            const visible = this.manager.visual.visible;
+            const index = visible.findIndex(visible => visible.chatWindowLocalId === this.localId);
+            if (index === -1) {
+                return undefined;
+            }
+            return index;
+        }
```

Comment:
unrelated code move is a bit annoying to review, might be worth a quick separate PR if we want to clean this up